ui/lib/config-ui/addon/components/messages/page/list.hbs

```diff
@@ -0,0 +1,113 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<Messages::TabPageHeader @authenticated={{@authenticated}}>
+  <:toolbarActions>
+    <ToolbarLink data-test-toolbar-create-message @route="messages.create" @type="add">
+      Create message
+    </ToolbarLink>
+  </:toolbarActions>
+</Messages::TabPageHeader>
+
+{{#if @messages.length}}
+  {{#each @messages as |message|}}
+    <LinkedBlock
+      data-test-list-item={{message.id}}
+      class="list-item-row"
+      @params={{array "messages.message.details" message.id}}
+      @linkPrefix={{this.mountPoint}}
+    >
+      <div class="level is-mobile">
+        <div class="level-left">
+          <div>
+            <Hds::Text::Display @tag="h2" data-linked-block-title={{message.id}}>
+              <Icon @name="message-circle" class="auto-width" aria-label="message" />
+              {{message.title}}
+            </Hds::Text::Display>
+            <div class="has-top-margin-xs">
+              <Hds::Badge
+                @text={{if
+                  message.active
+                  (if
+                    message.endTime
+                    (concat "Active until " (date-format message.endTime "MMM d, yyyy hh:mm aaa" withTimeZone=true))
+                    "Active"
+                  )
+                  (if
+                    message.isStartTimeAfterToday
+                    (concat "Scheduled: " (date-format message.startTime "MMM d, yyyy hh:mm aaa" withTimeZone=true))
+                    (concat "Inactive: " (date-format message.endTime "MMM d, yyyy hh:mm aaa" withTimeZone=true))
+                  )
+                }}
+                @color={{if message.active "highlight" "neutral"}}
+                data-test-badge={{concat "active:" message.active}}
+              />
+              <Hds::Badge
+                @text={{(capitalize message.type)}}
+                @color={{if message.active "highlight" "neutral"}}
+                data-test-badge={{message.type}}
+              />
+            </div>
+          </div>
+        </div>
+        <div class="level-right is-flex is-paddingless is-marginless">
+          <div class="level-item">
+            <PopupMenu @name="engine-menu">
+              <Confirm as |c|>
+                <nav class="menu">
+                  <ul class="menu-list">
+                    {{#if message.canEditCustomMessages}}
```

Comment:
does the permissions "update" also allow them to edit a message?

Reply:
Yup!